ce/ce/installers/espidf.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { platform } from 'os';
+import { delimiter } from 'path';
+import { log } from '../cli/styling';
+import { i } from '../i18n';
+import { InstallEvents } from '../interfaces/events';
+import { Session } from '../session';
+import { execute } from '../util/exec-cmd';
+import { isFilePath, Uri } from '../util/uri';
+import { Vcpkg } from '../vcpkg';
+
+export async function installEspIdf(session: Session, events: Partial<InstallEvents>, targetLocation: Uri) {
+  // create the .espressif folder for the espressif installation
+  await targetLocation.createDirectory('.espressif');
+  session.activation.addTool('IDF_TOOLS_PATH', targetLocation.join('.espressif').fsPath);
+
+  const vcpkg = new Vcpkg(session);
+  const pythonPath = await vcpkg.fetch('python3_with_venv');
+  if (!pythonPath) {
+    throw new Error(i`Could not activate esp-idf: python was not found.`);
+  }
+
+  const directoryLocation = await isFilePath(targetLocation) ? targetLocation.fsPath : targetLocation.toString();
+
+  const extendedEnvironment: NodeJS.ProcessEnv = {
+    ... await session.activation.getEnvironmentBlock(),
+    IDF_PATH: directoryLocation,
+    IDF_TOOLS_PATH: `${directoryLocation}/.espressif`
+  };
+
+  const installResult = await execute(pythonPath, [
+    `${directoryLocation}/tools/idf_tools.py`,
+    'install',
+    '--targets=all'
+  ], {
+    env: extendedEnvironment,
+    onStdOutData: (chunk) => {
+      const regex = /\s(100)%/;
+      chunk.toString().split('\n').forEach((line: string) => {
+        const match_array = line.match(regex);
+        if (match_array !== null) {
+          events.heartbeat?.('Installing espidf');
+        }
+      });
+    }
+  });
+
+  if (installResult.code) {
+    return false;
+  }
+
+  const installPythonEnv = await execute(pythonPath, [
+    `${directoryLocation}/tools/idf_tools.py`,
+    'install-python-env'
+  ], {
+    env: extendedEnvironment
+  });
+
+  if (installPythonEnv.code) {
+    return false;
+  }
+
+  return true;
```

Comment:
Elsewhere you made this:

```suggestion
  return installPythonEnv.code === 0;
```